Index: kwave-0.8.7-1/plugins/fileinfo/KeywordWidgetBase.ui
===================================================================
--- kwave-0.8.7-1.orig/plugins/fileinfo/KeywordWidgetBase.ui	2011-11-17 00:10:41.000000000 +1100
+++ kwave-0.8.7-1/plugins/fileinfo/KeywordWidgetBase.ui	2012-01-26 13:18:35.000000000 +1100
@@ -63,7 +63,7 @@
      </property>
      <property name="whatsThis">
       <string>&lt;b&gt;Auto-Generate Keywords&lt;/b&gt;&lt;br&gt;
-Create a list of keywords by going through all present file informations and merge them with the current list of keywords.</string>
+Create a list of keywords by going through all present file information and merge them with the current list of keywords.</string>
      </property>
      <property name="text">
       <string>A&amp;uto</string>
Index: kwave-0.8.7-1/plugins/playback/PlayBack-ALSA.cpp
===================================================================
--- kwave-0.8.7-1.orig/plugins/playback/PlayBack-ALSA.cpp	2011-11-17 00:10:36.000000000 +1100
+++ kwave-0.8.7-1/plugins/playback/PlayBack-ALSA.cpp	2012-01-26 13:18:35.000000000 +1100
@@ -567,7 +567,7 @@
 		break;
 	    case EBUSY:
 		reason = i18n(
-		"The device is busy. Maybe an other application is\n"\
+		"The device is busy. Maybe another application is\n"\
 		"currently using it. Please try again later.\n"\
 		"(Hint: you might find out the name and process ID of\n"\
 		"the program by calling: \"fuser -v %1\"\n"\
Index: kwave-0.8.7-1/plugins/playback/PlayBack-OSS.cpp
===================================================================
--- kwave-0.8.7-1.orig/plugins/playback/PlayBack-OSS.cpp	2011-11-17 00:10:36.000000000 +1100
+++ kwave-0.8.7-1/plugins/playback/PlayBack-OSS.cpp	2012-01-26 13:18:35.000000000 +1100
@@ -138,7 +138,7 @@
 		break;
 	    case EBUSY:
 		reason = i18n(
-		"The device is busy. Maybe an other application is\n"\
+		"The device is busy. Maybe another application is\n"\
 		"currently using it. Please try again later.\n"\
 		"(Hint: you might find out the name and process ID of\n"\
 		"the program by calling: \"fuser -v %1\"\n"\
Index: kwave-0.8.7-1/plugins/record/RecordDlg.ui
===================================================================
--- kwave-0.8.7-1.orig/plugins/record/RecordDlg.ui	2011-11-17 00:10:47.000000000 +1100
+++ kwave-0.8.7-1/plugins/record/RecordDlg.ui	2012-01-26 13:18:35.000000000 +1100
@@ -1053,7 +1053,7 @@
                   <string>Recording device</string>
                  </property>
                  <property name="whatsThis">
-                  <string>&lt;b&gt;Device:&lt;/b&gt;&lt;p&gt;This list shows a selection of some known recording devices suitable for the choosen recording method. If your device is not listed here, you can enter a device name on your own or search through the filesystem.&lt;/p&gt;</string>
+                  <string>&lt;b&gt;Device:&lt;/b&gt;&lt;p&gt;This list shows a selection of some known recording devices suitable for the chosen recording method. If your device is not listed here, you can enter a device name on your own or search through the filesystem.&lt;/p&gt;</string>
                  </property>
                  <property name="editable">
                   <bool>true</bool>
Index: kwave-0.8.7-1/po/cs.po
===================================================================
--- kwave-0.8.7-1.orig/po/cs.po	2011-11-27 01:58:14.000000000 +1100
+++ kwave-0.8.7-1/po/cs.po	2012-01-26 13:18:35.000000000 +1100
@@ -4011,7 +4011,7 @@
 "nového řádku!"
 
 #: libkwave/GenreType.cpp:144
-msgid "Psychadelic"
+msgid "Psychedelic"
 msgstr "Psychedelická hudba"
 
 #: libkwave/GenreType.cpp:172
Index: kwave-0.8.7-1/po/de.po
===================================================================
--- kwave-0.8.7-1.orig/po/de.po	2011-11-27 01:58:14.000000000 +1100
+++ kwave-0.8.7-1/po/de.po	2012-01-26 13:18:35.000000000 +1100
@@ -4013,8 +4013,8 @@
 "einem Punkt enden. Keine Zeilenumbrücke einfügen!"
 
 #: libkwave/GenreType.cpp:144
-msgid "Psychadelic"
-msgstr "Psychadelic"
+msgid "Psychedelic"
+msgstr "Psychedelic"
 
 #: libkwave/GenreType.cpp:172
 msgid "Psychedelic Rock"
Index: kwave-0.8.7-1/po/fr.po
===================================================================
--- kwave-0.8.7-1.orig/po/fr.po	2011-11-27 01:58:14.000000000 +1100
+++ kwave-0.8.7-1/po/fr.po	2012-01-26 13:18:35.000000000 +1100
@@ -4167,7 +4167,7 @@
 "N'utilisez pas de retour chariot !"
 
 #: libkwave/GenreType.cpp:144
-msgid "Psychadelic"
+msgid "Psychedelic"
 msgstr "Psychédélique"
 
 #: libkwave/GenreType.cpp:172
Index: kwave-0.8.7-1/po/kwave.pot
===================================================================
--- kwave-0.8.7-1.orig/po/kwave.pot	2011-11-27 01:58:14.000000000 +1100
+++ kwave-0.8.7-1/po/kwave.pot	2012-01-26 13:18:35.000000000 +1100
@@ -1000,7 +1000,7 @@
 msgstr ""
 
 #: libkwave/GenreType.cpp:144
-msgid "Psychadelic"
+msgid "Psychedelic"
 msgstr ""
 
 #: libkwave/GenreType.cpp:145
Index: kwave-0.8.7-1/libkwave/GenreType.cpp
===================================================================
--- kwave-0.8.7-1.orig/libkwave/GenreType.cpp	2011-11-17 00:14:30.000000000 +1100
+++ kwave-0.8.7-1/libkwave/GenreType.cpp	2012-01-26 13:22:55.000000000 +1100
@@ -141,7 +141,7 @@
 	    i18n("Native American"),
 	    i18n("Cabaret"),
 	    i18n("New Wave"),
-	    i18n("Psychadelic"),
+	    i18n("Psychedelic"),
 	    i18n("Rave"),
 	    i18n("Showtunes"),
 	    i18n("Trailer"),
